### PR TITLE
Support for `infrastructure_encryption_required` in `azurerm_storage_encryption_scope`

### PR DESCRIPTION
**Pull request by yupwei68**
All tests have passed.

Fixes: https://github.com/terraform-providers/terraform-provider-azurerm/issues/11463

### CODE REVIEW

File: azurerm/internal/services/storage/storage_encryption_scope_resource.go [hunk 1]
```diff
@@ -67,6 +67,12 @@ func resourceStorageEncryptionScope() *schema.Resource {
 				Optional:     true,
 				ValidateFunc: keyVaultValidate.KeyVaultChildID,
 			},
+
+			"require_infrastructure_encryption": {
```
Comment (katbyte) [inline azurerm/internal/services/storage/storage_encryption_scope_resource.go:71]: for consistency with the tenses used in the provider
```suggestion
			"infrastructure_encryption_required": {
```

**Review by katbyte — CHANGES_REQUESTED** [review]
looks good aside from one comment

**Comment by yupwei68** [comment]
Thanks for your comments kt. All tests have passed. Please continue reviewing.

**Review by katbyte — APPROVED** [review]
Thanks, LGTM 🤖

**Pull request merged**